shared/oae/api/oae.api.util.js

```diff
@@ -1804,14 +1804,25 @@ define(['exports', 'require', 'jquery', 'underscore', 'oae.api.config', 'markdow
             window.location = '/servermaintenance';
         };
 
+        /**
+         * TODO bla bla bla
+         */
+        var logInThroughAnotherTenant = function(tenant) {
+            let relative = url().attr('query').slice(4);
+            let host = url().attr('base').replace(/(\w*)\./, tenant.alias + '.');
+            let newURL = host + relative;
+            window.location = newURL;
```

Comment:
Will this continue to work when the tenants are on different domains (for example, *.unity.ac to *.oaeproject.org)?

Reply:
hmm no, I don't think so. Which means I might need to do some changes such as storing not just the tenantAlias but also the tenant domain 

🖐

Reply:
OK, so I did a bit of research on this and the reason why the scenario won't work is because of the way cookies work. If user logs in to `tenantA.unity.ac`, the cookie domains one can set are either `.tenantA.unity.ac` and `.unity.ac`. The latter is the one that's useful to us, so that other tenants can access it (e.g. `tenantB.unity.ac`). But one could never set a cookie for `tenantC.oaeproject.org` to see, because the domains are different. (Read [this thread](https://stackoverflow.com/questions/1062963/how-do-browser-cookie-domains-work) for more info.)

So that means this is indeed a limitation of this approach, and I'm guessing that all other kinds of browser storage respect the same philosophy for security reasons.

Reply:
Fair enough, I hadn't thought of that. That's an unavoidable limitation then, but I guess generally most tenancies in any instance would be using the same main domain.